Use env logger

Flat opened the pull request:
Replace `panic!` and `eprintln!` to use `error!` `info!` and `debug!` from `env_logger` - which by default all print to stderr. Closes #3

Commit:
Implement env_logger

Replace panic! and eprintln! to use error! info! and debug! from env_logger - which by default all print to stderr. Closes #3

Commit:
Fix whitespace

dsprenkels commented on Cargo.toml:
```diff
@@ -12,6 +12,8 @@ name = "secret-share-combine"
 
 [dependencies]
+log = "0.3.8"
+env_logger = "0.4.3"
```
I think I would like to support all the patch versions of these crates. Could this be changed to the following?

```
log = "0.3"
env_logger = "0.4"
```

dsprenkels commented on src/bin/combine.rs:
```diff
@@ -19,19 +23,24 @@ fn argparse<'a>() -> ArgMatches<'a> {
         .read_to_string(&mut shares_string)
-        .unwrap_or_else(|err| panic!("Error while reading stdin: {}", err));
+        .unwrap_or_else(|err| { error!("Error while reading stdin: {}", err);
+                                exit(1)});
```
Little nit: I think this would me more readable if we inserted an extra newline:

```rust
        .unwrap_or_else(|err| {
            error!("Error while reading stdin: {}", err);
            exit(1)
        });
```

dsprenkels commented:
Thank you so much for you contribution! I have a couple of little nits, but that should be all.

dsprenkels commented on src/bin/combine.rs:
```diff
@@ -78,8 +94,14 @@ fn main() {
+        Ok(None) => {
+            error!("Shares did not combine to a valid secret");
+            exit(1);
+            },
```
Nit: Unindent this brace.

dsprenkels commented on src/bin/combine.rs:
```diff
@@ -78,8 +94,14 @@ fn main() {
+        Err(err) => {
+            error!("Error while combining shares: {}", err);
+            exit(1);
+            },
```
Nit: Also unindent this brace.

dsprenkels commented on src/bin/combine.rs:
```diff
@@ -91,8 +113,8 @@ fn main() {
-            eprintln!("Warning: Invalid utf-8 text, some symbols may be lost!");
-            eprintln!("Note: The hex representation of the secret is '{}'.", hex);
+            info!("Warning: Invalid utf-8 text, some symbols may be lost!");
+            debug!("Note: The hex representation of the secret is '{}'.", hex);
```
I think `info!` may not be explicit enough to the user. Maybe change this to `warn!` and `info!`?

Commit:
Requested changes for PR #4

Change log and env_logger crates to not specify patch version. Fix indentation on two end brackets. Change the use of `info!` and `debug!` to `warn!` and `info!` in the match statement from String::from_utf8(secret)

dsprenkels commented:
Thank you! :smile:

Flat commented:
Glad I could contribute!
